pyomo/solvers/plugins/solvers/GUROBI.py

```diff
@@ -69,14 +72,29 @@ def __new__(cls, *args, **kwds):
         if mode == 'os':
             opt = SolverFactory('_ossolver', **kwds)
         elif mode == 'nl':
-            opt = SolverFactory('asl', **kwds)
+            opt = SolverFactory('_gurobi_nl', **kwds)
         else:
             logger.error('Unknown IO type: %s' % mode)
             return
         opt.set_options('solver=gurobi_ampl')
         return opt
 
 
+@SolverFactory.register('_gurobi_nl', doc='NL interface to the Gurobi solver')
+class GUROBINL(ASL):
+    """NL interface to gurobi_ampl."""
+    def license_is_valid(self):
+        m = ConcreteModel()
+        m.x = Var(bounds=(0,1))
+        m.obj = Objective(expr=m.x)
+        try:
+            with capture_output():
+                self.solve(m)
+            return m.x.value == 0
```

Comment:
Is there any risk this does the wrong thing if `x` isn't exactly 0?

Reply:
There is a risk, but this is *such* a simple problem that Gurobi should solve it exactly in presolve.